devices/tuya.js

```diff
@@ -1969,40 +1969,89 @@ module.exports = [
             exposes.text('schedule_friday', ea.STATE),
             exposes.text('schedule_saturday', ea.STATE),
             exposes.text('schedule_sunday', ea.STATE),
-            exposes.binary('online', ea.STATE_SET, 'ON', 'OFF').withDescription('Is the device online'),
+            exposes.binary('online', ea.STATE_SET, 'ON', 'OFF').withDescription('Turn on this property to poll current data from the ' +
+                'device. It can be used to periodically fetch a new local temperature since the device doesn\'t update itself. ' +
+                'Setting this property doesn\'t turn on the display.'),
             exposes.numeric('error_status', ea.STATE).withDescription('Error status'),
         ],
+        meta: {
+            tuyaDatapoints: [
+                [2, 'preset', tuya.valueConverterBasic.lookup({'auto': 0, 'manual': 1, 'holiday': 3}),
+                    {datatype: tuya.sendDataPointEnum}],
+                [8, 'open_window', tuya.valueConverter.onOff],
+                [10, 'frost_protection', tuya.valueConverter.onOff],
+                [16, 'current_heating_setpoint', tuya.valueConverter.divideBy10],
+                [27, 'local_temperature_calibration', tuya.valueConverter.localTempCalibration],
+                [31, 'working_day', tuya.valueConverterBasic.lookup({'mon_sun': 0, 'mon_fri+sat+sun': 1, 'separate': 2}),
+                    {datatype: tuya.sendDataPointEnum}],
+                [32, 'holiday_temperature', tuya.valueConverter.divideBy10],
+                [35, 'battery', tuya.valueConverter.raw],
+                [40, 'child_lock', tuya.valueConverter.lockUnlock],
+                [45, 'error_status', tuya.valueConverter.raw],
+                [46, 'holiday_start_stop', tuya.valueConverter.thermostatHolidayStartStop],
+                [101, 'boost_timeset_countdown', tuya.valueConverter.raw],
+                [102, 'open_window_temperature', tuya.valueConverter.divideBy10],
+                [104, 'comfort_temperature', tuya.valueConverter.divideBy10],
+                [105, 'eco_temperature', tuya.valueConverter.divideBy10],
+                [106, 'schedule', tuya.valueConverter.thermostatScheduleDay],
+                [107, 'system_mode', tuya.valueConverterBasic.lookup({'heat': false, 'off': true})],
+                [115, 'online', tuya.valueConverter.onOff],
+                [108, 'schedule_monday', tuya.valueConverter.thermostatScheduleDay],
+                [112, 'schedule_tuesday', tuya.valueConverter.thermostatScheduleDay],
+                [109, 'schedule_wednesday', tuya.valueConverter.thermostatScheduleDay],
+                [113, 'schedule_thursday', tuya.valueConverter.thermostatScheduleDay],
+                [110, 'schedule_friday', tuya.valueConverter.thermostatScheduleDay],
+                [114, 'schedule_saturday', tuya.valueConverter.thermostatScheduleDay],
+                [111, 'schedule_sunday', tuya.valueConverter.thermostatScheduleDay],
```

Comment:
Checking with @bwindsor  (https://github.com/Koenkk/zigbee-herdsman-converters/pull/4735) 

It would be good to do some unification between the different TuYa thermostat schedule formats that are available now. But from what I see here the structure is different; Moes BHT-002 uses 1 datapoint and this thermostat uses multiple

Reply:
I think it's necessary to create several unified converters, because some devices have a similar data format, while others have a completely different data format.
For example, these converters can be combined into one with params to configure such things like the number of periods, temperature converting, etc:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/8df9506531ab2dab43174a826d58ecdb016a0c6c/converters/toZigbee.js#L3453-L3463
https://github.com/Koenkk/zigbee-herdsman-converters/blob/8df9506531ab2dab43174a826d58ecdb016a0c6c/converters/toZigbee.js#L5106-L5119
https://github.com/Koenkk/zigbee-herdsman-converters/blob/9379ccbad81a2087097ca3141ef572f6660f9487/devices/tuya.js#L253-L267
For example, them can be combined to function with the next params
`tuya.valueConverter.tuyaThermostatSchedule1({ periods: 12, convertTemp: (t) => t * 2 })`

Some converters look completely different, e.g.:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/8df9506531ab2dab43174a826d58ecdb016a0c6c/converters/toZigbee.js#L4811-L4822
It's difficult to combine with the previous ones.

Function chaining can be a way out. It can be created many little function-utils that transform data into the required form. Something like this, declarative form:
`tuya.valueConverter.split(' ').schedule({ periods: 4, periodSize: 3 }).condition({ temperature: { min: 5, max: 35 } }).convert({ temperature: (t) => t * 2 });`
And the schedule method should return a strict object, so it can be easily used in the next method e.g.:
`{ p1: { hour: 10, minute: 20, temperature: 20.5 }, p2: ... }`
But it seems a bit complicated to me, I don't think it's a good idea.